### PR TITLE
A bunch of improvements

### PR DESCRIPTION
**Pull request by dhylands**
* fixes issues discovered with Cree bulbs
* adds support for Samsung SmartThings smartplugs
* improves a bunch of things around config reporting

### CODE REVIEW

**Comment by benfrancis** [comment]
> adds support for Samsung SmartThings smartplugs

Out of interest what changes were made? The UK version of the SmartThings Outlet already worked for me.

**Comment by dhylands** [comment]
It was more a case of making things work better. The specific things that were changed were:
* Reduced the reporting frequency of attributes. With the previous settings, very small changes in frequency, current, voltage and power were causing all of these to be reported every second.
* Removed the Level slider (which was non-functional anyways).

File: zb-adapter.js [hunk 1]
```diff
@@ -676,7 +750,7 @@ class ZigbeeAdapter extends Adapter {
           const devInfoNode = devInfo.nodes[nodeId];
           let node = this.nodes[nodeId];
           if (!node) {
-            node = new ZigbeeNode(this, devInfoNode.addr64, devInfoNode.addr16);
+            node = new ZigbeeNode(this, devInfoNode.addr64, 'fffe');
```
[the remark this answers is not in the document]
Comment (dhylands) [reply zb-adapter.js]: 'fffe' is the unknown 16-bit address. I'll add a constant

File: zb-property.js [hunk 2]
```diff
@@ -153,19 +166,18 @@ class ZigbeeProperty extends Property {
...
-          hue = attrEntry.attrData;
-          break;
-        case 1:
-          sat = attrEntry.attrData;
-          break;
-      }
-    }
+  parseColorAttr(attrEntry) {
+    if (attrEntry.attrId == ATTR_ID_LIGHTINGCOLORCTRL_CURRENTHUE) {
+      // We expect that we'll always get the hue in one call, and
```
[the remark this answers is not in the document]
Comment (dhylands) [reply zb-property.js]: We do a read request with hue specified before saturation, and the readRsp comes back with the attributes in the same order.

The spec says:
```
The attribute identifier field is 16 bits in length and SHALL contain the identifier
of the attribute that has been read (or of which an element has been read).
This field SHALL contain the same value that was included in the corresponding
attribute identifier field of the original Read Attributes or Read Attributes
Structured command.
```
It doesn't seem to say anything about the order, but all of the devices I've seen always return the attributes in the readRsp in the same order as they appear in the read command.

**Pull request merged**